src/Ocelot/LoadBalancer/LoadBalancers/RoundRobin.cs

```diff
@@ -27,6 +24,12 @@ public async Task<Response<ServiceHostAndPort>> Lease(HttpContext httpContext)
             var services = await _services();
             lock (_lock)
             {
+                if (services.Count < 1)
+                {
+                    // When the downstream service is not found, LeastConnection prompts Warn and RoundRobin throws an exception, eg: /favicon.ico
+                    return new ErrorResponse<ServiceHostAndPort>(new ServicesAreEmptyError($"Services were empty for {httpContext.Request.Path}"));
```

Comment:
Does this error returning fix the bug with exception generation by global exception handler?
How will `ErrorResponse` prevent exception generating in upper contexts?